pkg/lib/telemetry/telemetry.go

```diff
@@ -17,121 +17,190 @@ limitations under the License.
 package telemetry
 
 import (
-	"bytes"
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
 	"time"
 
 	"github.com/cortexlabs/cortex/pkg/consts"
 	"github.com/cortexlabs/cortex/pkg/lib/errors"
+	"github.com/cortexlabs/cortex/pkg/lib/parallel"
+	"github.com/getsentry/sentry-go"
+	"gopkg.in/segmentio/analytics-go.v3"
 )
 
-const (
-	eventPath = "/events"
-	errorPath = "/errors"
-)
+var _segment analytics.Client
+var _config *Config
 
-type Client struct {
-	url             string
-	http            http.Client
-	EnableTelemetry bool
-	OperatorID      string
+type Config struct {
+	Enabled         bool
+	UserID          string
+	Environment     string
+	ShouldLogErrors bool
 }
 
-func New(telemetryURL string, hashedAccountID string, enable bool) *Client {
-	timeout := time.Duration(10 * time.Second)
-	httpClient := http.Client{
-		Timeout: timeout,
-	}
-	telem := &Client{
-		url:             telemetryURL,
-		http:            httpClient,
-		EnableTelemetry: enable,
-		OperatorID:      hashedAccountID,
-	}
+type silentSegmentLogger struct{}
+
+func (logger silentSegmentLogger) Logf(format string, args ...interface{}) {
+	return
+}
 
-	return telem
+func (logger silentSegmentLogger) Errorf(format string, args ...interface{}) {
+	return
 }
 
-type UsageEvent struct {
-	Timestamp  time.Time `json:"timestamp"`
-	Version    string    `json:"version"`
-	OperatorID string    `json:"operator_id"`
-	Event      string    `json:"event"`
+type silentSentryLogger struct{}
+
+func (logger silentSentryLogger) Write(p []byte) (n int, err error) {
+	return len(p), nil
 }
 
-func (c *Client) sendUsageEvent(name string) {
-	usageEvent := UsageEvent{
-		Timestamp:  time.Now(),
-		Version:    consts.CortexVersion,
-		OperatorID: c.OperatorID,
-		Event:      name,
+func Init(telemetryConfig Config) error {
+	if !telemetryConfig.Enabled {
+		_config = nil
+		return nil
+	}
+
+	if telemetryConfig.UserID == "" {
+		return errors.New("user ID must be specified to enable telemetry")
 	}
 
-	byteArray, _ := json.Marshal(usageEvent)
-	resp, err := c.http.Post(c.url+eventPath, "application/json", bytes.NewReader(byteArray))
+	err := sentry.Init(sentry.ClientOptions{
+		Dsn:         "https://5cea3d2d67194d028f7191fcc6ebca14@sentry.io/1825326",
+		Release:     consts.CortexVersion,
+		Environment: telemetryConfig.Environment,
+	})
 	if err != nil {
-		errors.PrintError(err)
-		return
+		_config = nil
+		return err
 	}
 
-	defer resp.Body.Close()
+	var segmentLogger analytics.Logger
+	if !telemetryConfig.ShouldLogErrors {
+		sentry.Logger.SetOutput(silentSentryLogger{})
+		segmentLogger = silentSegmentLogger{}
+	}
 
-	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		if byteArray, err := ioutil.ReadAll(resp.Body); err == nil {
-			fmt.Println(string(byteArray))
-		}
+	_segment, err = analytics.NewWithConfig("0WvoJyCey9z1W2EW7rYTPJUMRYat46dl", analytics.Config{
```

Comment:
I think this value and the Dsn link should be moved to consts package or declared at the top of the file?